app/assets/javascripts/govuk_publishing_components/components/reorderable-list.js

```diff
@@ -84,10 +99,15 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
   }
 
   ReorderableList.prototype.updateOrderIndexes = function () {
-    var $orderInputs = this.$module.querySelectorAll('.gem-c-reorderable-list__actions input')
-    for (var i = 0; i < $orderInputs.length; i++) {
-      $orderInputs[i].setAttribute('value', i + 1)
-    }
+    var $actions = this.$module.querySelectorAll('.gem-c-reorderable-list__actions')
+
+    $actions.forEach(function ($action, index) {
+      $action.querySelector('input').value = index + 1
+
+      $action.querySelectorAll('button[data-ga4-event]').forEach(function ($trackButton) {
+        $trackButton.dataset.indexSection = index
+      })
+    })
```

Comment:
I have a question about indexing after reordering items. It seems that when I move the first item to the second position, the following values appear for the first item (which now appears as the second item in the list):

- The input (index) `value` is `1`.
- `trackButton.dataset.indexSection` is also `1`.

However, the `data-ga4-event` attribute on both buttons show:

```json
{
  "event_name": "select_content",
  "type": "reorderable list",
  "section": "Budget 2018",
  "index_section": 0,
  "index_section_count": 5,
  "action": "Up"
}
```
So, `index_section` remains 0, which appears to have the item's original index, not its updated position.

Is that correct?

Reply:
Thanks, Patrick - that makes sense. I agree; otherwise, you'd have to parse the JSON, which seems like unnecessary extra work